nightfall-deployer/src/circuit-setup.mjs

```diff
@@ -124,21 +127,27 @@ async function setupCircuits() {
       const vkArray = Object.values(vk).flat(Infinity); // flatten the Vk array of arrays because that's how Key_registry.sol likes it.
       const folderpath = circuit.slice(0, -4); // remove the .zok extension
 
-      let tx;
-      if (config.USE_STUBS) {
-        tx = keyRegistry.methods.registerVerificationKey(
-          vkArray,
-          config.VK_IDS[folderpath.slice(0, -5)],
-        );
-      } else {
-        tx = keyRegistry.methods.registerVerificationKey(vkArray, config.VK_IDS[folderpath]);
-      }
+      const call = keyRegistry.methods.registerVerificationKey(
+        vkArray,
+        config.VK_IDS[USE_STUBS ? folderpath.slice(0, -5) : folderpath],
+      );
 
-      // when deploying on infura - do serial tx execution to avoid nonce issue
       // when using a private key, we shouldn't assume an unlocked account and we sign the transaction directly
       if (config.ETH_PRIVATE_KEY) {
-        await Web3.submitRawTransaction(await tx.encodeABI(), keyRegistryAddress);
-      } else await tx.send();
+        const tx = {
+          from: process.env.FROM_ADDRESS,
+          to: keyRegistry.options.address,
+          data: call.encodeABI(),
```

Comment:
Explain why you do this in the code

Reply:
There's no account management on edge, you can't use send(). You need to encodeABI() and then sign it with a supplied private key

Reply:
Again, the idea is that you highlight this in the code so that future readers understand. We don't want somebody removing this code in the future, and its good idea to have this explained in comments in the code.

Reply:
absolutely, I'll add another commit with a comment